packages/imperative/src/rest/src/client/Proxy.ts

```diff
@@ -0,0 +1,94 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { env } from 'process';
+import { URL } from 'url';
+import { Agent } from 'https';
+import { HttpProxyAgent } from 'http-proxy-agent';
+import { HttpsProxyAgent } from 'https-proxy-agent';
+
+import { HTTP_PROTOCOL_CHOICES, HTTP_PROTOCOL, HTTPS_PROTOCOL } from '../session/SessConstants';
+import { ISession } from '../session/doc/ISession';
+
+export class Proxy {
+
```

Comment:
When adding a new family of functions in a class, we try to have a block comment for each function describing its purpose, parameters, and return value. Perhaps when you have finalized the details of the implementation, you could add such comments within the new Proxy class.

Reply:
@gejohnston thanks for the review and sorry for the delay. I now added a commit with the jsdoc comments following the pattern I saw other modules use in the folder.